src/mapreduce.jl

```diff
@@ -285,3 +285,43 @@ end
         @inbounds return similar_type(a, T, Size($Snew))(tuple($(exprs...)))
     end
 end
+
+struct _InitialValue end
+
+_maybeval(dims::Integer) = Val(Int(dims))
+_maybeval(dims) = dims
+_valof(::Val{D}) where D = D
```

Comment:
I'm kind of surprised this doesn't exist in `Base` and I'm somewhat tempted to submit a PR to make it exist. Any thoughts on what it should be called though?

Reply:
> I'm kind of surprised this doesn't exist in `Base` and I'm somewhat tempted to submit a PR to make it exist.

Yeah, I define this all the time and it'd be great if `Base` has this.

> Any thoughts on what it should be called though?

Maybe `v[]`?  Or `get(v)`?

Reply:
Ref https://github.com/JuliaLang/julia/issues/34758